### PR TITLE
Add Lookup JSON String output to test, make test util

### PR DESCRIPTION
**Pull request by ryankwagner**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

- Generate JSON parser for a dummy query using MAHA_LOOKUP.
- Create a MAHA_LOOKUP dummy class to ensure users use uniform params.
- Create a util so these functions can be reused.
- Add input column optional mapping
- Tests assert on some bad input cases

### CODE REVIEW

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/BaseRocksDBLookupExtractor.java [hunk 1]
```diff
@@ -122,6 +125,10 @@ public String apply(@NotNull String key, @NotNull String valueColumn, DecodeConf
                     LOG.error("RocksDB instance is null");
                     LOG.error("Failed to get lookup value from cache. Falling back to lookupService.");
                     serviceEmitter.emit(ServiceMetricEvent.builder().build(MonitoringConstants.MAHA_LOOKUP_GET_CACHE_VALUE_FAILURE + "_" + extractionNamespace.getNamespace(), 1));
+                    String cacheByteString = new String(lookupService.lookup(new LookupService.LookupData(extractionNamespace, key, valueColumn, decodeConfigOptional)));
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/BaseRocksDBLookupExtractor.java:128]: We do not have to map the result from Lookup Service. Lookup service calls endpoints on peer and will already returned the mapped value. Remapping is not needed.

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/BaseRocksDBLookupExtractor.java [hunk 2]
```diff
@@ -140,6 +147,10 @@ public String apply(@NotNull String key, @NotNull String valueColumn, DecodeConf
                     }
                     return null;
                 } else {
+                    String cacheByteString = new String(cacheByteValue);
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/BaseRocksDBLookupExtractor.java:150]: can we trim the cacheByteValue before mapping?

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/MahaLookupOperatorConversion.java [hunk 3]
```diff
@@ -66,9 +67,15 @@ public DruidExpression toDruidExpression(PlannerContext plannerContext, RowSigna
                     final Expr columnName = inputExpressions.get(2).parse(plannerContext.getExprMacroTable()); // maha lookup name
 
                     LookupReferencesManager lrm = (LookupReferencesManager) lookupReferencesManager;
-                    String missingValue = getMissingValue(inputExpressions, plannerContext);
-
-                    if (arg.isSimpleExtraction() && lookupName.isLiteral() && columnName.isLiteral()) {
+                    String missingValue = getMissingValue(inputExpressions, plannerContext, 3, MISSING_VALUE);
+                    String extractionMap = getMissingValue(inputExpressions, plannerContext, 4, "-3");
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/MahaLookupOperatorConversion.java:71]: need to write new wrapper function to get these optional maps.

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/MahaLookupOperatorConversion.java [hunk 4]
```diff
@@ -66,9 +67,15 @@ public DruidExpression toDruidExpression(PlannerContext plannerContext, RowSigna
...
+                    String valueExtractionMap = getMissingValue(inputExpressions, plannerContext, 5, "-3");
+                    //TODO: Enhance by passing in KV separator & delimeter.
+                    //Also, allow passing of Map type statements: Case, KV pair, etc. ex: CASE, MAP(',','->')
+                    Map<String, String> dimensionOverrideMap = extractionMap == null || extractionMap.equals("-3") ? null : Splitter.on(",").withKeyValueSeparator("->").split(extractionMap);
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/MahaLookupOperatorConversion.java:75]: Good that we are using Guava for string to map. Makes life easy :) 
We can move separator and map separator to static constant value in this class.

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/MahaLookupOperatorConversion.java [hunk 5]
```diff
@@ -30,7 +31,7 @@
 
     private static final SqlFunction SQL_FUNCTION = OperatorConversions
             .operatorBuilder(DRUID_FUNC_NAME)
-            .operandTypes(SqlTypeFamily.ANY, SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER)
+            .operandTypes(SqlTypeFamily.ANY, SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER, SqlTypeFamily.CHARACTER)
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/MahaLookupOperatorConversion.java:34]: secondaryColOverrideMap is commonly used feature, we can move it right after missing value and push KeyOverrideMap  to last.

**Comment by pranavbhole** [comment]
@ryankwagner  Thank you for your PR, this pr was deployed to Sandbox and it works. You can address the comments and we can let this in ! 💯 🥇 🥇🥇🥇🥇🥇🥇🥇🥇🥇🥇🥇🥇🥇🥇

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/MahaLookupOperatorConversion.java (hunk 5 above)
Comment (ryankwagner) [reply druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/MahaLookupOperatorConversion.java:34]: You only want it moved around inside the lookup ordering itself, but retain the same ordering on all other existing lookups (non-sql), correct?

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/BaseRocksDBLookupExtractor.java (hunk 1 above)
Comment (ryankwagner) [reply druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/BaseRocksDBLookupExtractor.java:128]: It looks like LookupService here only works on a simple decodeConfig which can only match a single value (not a whole Map of values).

File: druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/BaseRocksDBLookupExtractor.java [hunk 6]
```diff
@@ -122,6 +125,10 @@ public String apply(@NotNull String key, @NotNull String valueColumn, DecodeConf
                     LOG.error("RocksDB instance is null");
                     LOG.error("Failed to get lookup value from cache. Falling back to lookupService.");
                     serviceEmitter.emit(ServiceMetricEvent.builder().build(MonitoringConstants.MAHA_LOOKUP_GET_CACHE_VALUE_FAILURE + "_" + extractionNamespace.getNamespace(), 1));
+                    String cacheByteString = new String(lookupService.lookup(new LookupService.LookupData(extractionNamespace, key, valueColumn, decodeConfigOptional))).trim();
+                    if (secondaryColOverrideMap != null && secondaryColOverrideMap.containsKey(cacheByteString)) {
+                        return Strings.emptyToNull(secondaryColOverrideMap.get(cacheByteString));
+                    }
```
Comment (pranavbhole) [inline druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/BaseRocksDBLookupExtractor.java:131]: In this case :  A->A_MAPPED, B->B_MAPPED.
LookupService will return the return A_MAPPED.

String cacheByteString = new String(lookupService.lookup(new LookupService.LookupData(extractionNamespace, key, valueColumn, decodeConfigOptional))).trim();
                    if (secondaryColOverrideMap != null && secondaryColOverrideMap.containsKey(cacheByteString)) {
                        return Strings.emptyToNull(secondaryColOverrideMap.get(cacheByteString));
                    }

thus we do not need above code, just return the value that you got from lookup service.

**Review by pranavbhole — APPROVED** [review]

**Pull request merged**